azurelinuxagent/common/utils/networkutil.py

```diff
@@ -157,56 +157,41 @@ def __get_firewalld_base_command(command):
 
     @staticmethod
     def __get_common_command_params(command, destination):
-        return ["-t", "security", command, "OUTPUT", "-d", destination, "-p", "tcp", "-m"]
+        return ["-t", "security", command, "OUTPUT", "-d", destination, "-p", "tcp"]
 
     @staticmethod
-    def __get_common_accept_command_params(wait, command, destination, uid):
-        cmd = AddFirewallRules.__get_common_command_params(command, destination)
-        cmd.extend(["owner", "--uid-owner", str(uid), "-j", "ACCEPT"])
-        return AddFirewallRules._add_wait(wait, cmd)
-
-    @staticmethod
-    def __get_common_drop_command_params(wait, command, destination):
-        cmd = AddFirewallRules.__get_common_command_params(command, destination)
-        cmd.extend(["conntrack", "--ctstate", "INVALID,NEW", "-j", "DROP"])
-        return AddFirewallRules._add_wait(wait, cmd)
-
-    @staticmethod
-    def get_accept_tcp_rule(command, destination, firewalld_command="", wait=""):
-        # This rule allows DNS TCP request to wireserver ip for non root users
-
+    def __get_firewall_base_command(firewalld_command="", wait=""):
+        # Firewalld.service fails if we set `-w` in the iptables command, so not adding it at all for firewalld commands
         if firewalld_command != "":
             cmd = AddFirewallRules.__get_firewalld_base_command(firewalld_command)
         else:
             cmd = AddFirewallRules.__get_iptables_base_command(wait)
-        cmd = cmd + ['-t', 'security', command, 'OUTPUT', '-d', destination, '-p', 'tcp', '--destination-port', '53', '-j', 'ACCEPT']
-        return cmd
 
-    @staticmethod
-    def get_iptables_accept_command(wait, command, destination, owner_uid):
-        cmd = AddFirewallRules.__get_iptables_base_command()
-        cmd.extend(AddFirewallRules.__get_common_accept_command_params(wait, command, destination, owner_uid))
         return cmd
 
     @staticmethod
-    def get_iptables_drop_command(wait, command, destination):
-        cmd = AddFirewallRules.__get_iptables_base_command()
-        cmd.extend(AddFirewallRules.__get_common_drop_command_params(wait, command, destination))
+    def get_accept_tcp_rule(command, destination, firewalld_command="", wait=""):
+        # This rule allows DNS TCP request to wireserver ip for non root users
+        cmd = AddFirewallRules.__get_firewall_base_command(firewalld_command, wait)
+
+        cmd.extend(AddFirewallRules.__get_common_command_params(command, destination))
```

Comment:
NIT: This call is common for all 3 rules, why not move it to the common function `__get_firewall_base_command`?